lib/preprocessor.cpp

```diff
@@ -114,7 +114,7 @@ static void inlineSuppressions(const simplecpp::TokenList &tokens, Settings &mSe
         // Add the suppressions.
         for (Suppressions::Suppression &suppr : inlineSuppressions) {
             suppr.fileName = relativeFilename;
-            suppr.lineNumber = tok->location.line-2;
+            suppr.lineNumber = tok->location.line-1;
```

Comment:
because `tok` is not the comment. The `tok` is the code below the comment. Imagine:

    void foo() {
        return; // cppcheck-suppress aaa
    }

Here `tok` will be the token for `}`.

Reply:
so I think we need to keep this code as it is to have old suppressions working.

When a suppression comment is found and there is code before it on the same line.. then a corresponding `Suppression` can be added directly.

Reply:
ok, I need more think.

Reply:
I got it.